src/transformation/index.ts

```diff
@@ -0,0 +1,72 @@
+import * as ts from "typescript";
+import * as tstl from "../LuaAST";
+import { LuaLibFeature } from "../LuaLib";
+import { setIfMissing } from "../utils";
+import { TransformationContext, TransformerPlugin, VisitorMap } from "./context";
+import { standardPlugins } from "./transformers";
+import { TranspileError } from "./utils/errors";
+import { getUsedLuaLibFeatures } from "./utils/lualib";
+
+export { TransformerPlugin } from "./context";
+
+const transpileErrorDiagnostic = (error: TranspileError): ts.Diagnostic => ({
+    file: error.node.getSourceFile(),
+    start: error.node.getStart(),
+    length: error.node.getWidth(),
+    category: ts.DiagnosticCategory.Error,
+    code: 0,
+    source: "typescript-to-lua",
+    messageText: error.message,
+});
+
+export function createVisitorMap(customPlugins: TransformerPlugin[]): VisitorMap {
+    const visitorMap: VisitorMap = new Map();
+    for (const plugin of [...standardPlugins, ...customPlugins]) {
+        for (const [syntaxKindKey, visitor] of Object.entries(plugin.visitors)) {
+            if (!visitor) continue;
+
+            const syntaxKind = Number(syntaxKindKey) as ts.SyntaxKind;
```

Comment:
This number cast seems strange, is it because keys are typed as strings?

Reply:
Yes. It also can be written as `ts.SyntaxKind[ts.SyntaxKind[syntaxKindKey]]` but that looks even more confusing